File: js/matrixRegionSum.js
function matrixRegionSum (matrix, topLeftCoords, bottomRightCoords) {
  const startCol = topLeftCoords[0]
       ,startRow = topLeftCoords[1]
       ,endCol = bottomRightCoords[0]
       ,endRow = bottomRightCoords[1];

  let sum = 0;
  for (let i = startRow; i <= endRow; i++) {
    for (let j = startCol; j <= endCol; j++) {
      sum += matrix[i][j];
    }
  }

  return sum;
}

let matrix = [
  [1,2,3],
  [4,5,6],
  [7,8,9]
]

console.log(matrixRegionSum(matrix, [0,0], [1,1]));

File: ruby/matrix_region_sum.rb
def matrix_region_sum(matrix, top_left_coords, bottom_right_coords)
  sum = 0

  (top_left_coords[0]..bottom_right_coords[0]).each do |i|
    (top_left_coords[1]..bottom_right_coords[1]).each do |j|
      sum += matrix[i][j]
    end
  end

  sum
end
